### PR TITLE
JAMES-2997 step #7 Do not rely on properties to determine if a messag…

### PR DESCRIPTION
**Pull request by chibenwa**
…e has attachment

Use attachment metadata directly

Extract a subpart of https://github.com/linagora/james-project/pull/3061

### CODE REVIEW

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java [hunk 1]
```diff
@@ -353,38 +363,6 @@ void hasAttachmentsShouldReturnFalseWhenPropertyIsPresentButFalse() throws IOExc
             .thenReturn(MESSAGE_UID);
         when(mailboxMessage.getModSeq())
             .thenReturn(ModSeq.first());
-        when(mailboxMessage.getProperties())
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java:356]: should you change the test name?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java [hunk 2]
```diff
@@ -347,7 +347,7 @@ void hasAttachmentsShouldReturnTrueWhenNonInlined() throws IOException {
     }
 
     @Test
-    void hasAttachmentsShouldReturnFalseWhenPropertyIsPresentButFalse() throws IOException {
+    void hasAttachmentsShouldReturnFalseWhenEmptyAttachmentsButFalse() throws IOException {
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java:350]: trailing `butFalse` still make sense?

**Comment by chibenwa** [comment]
` ActiveMQMailQueueTest.clearShouldNotFailWhenBrowsingIterating » NoSuchElement`

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**